src/ops/stream_events.rs

```diff
@@ -0,0 +1,101 @@
+//! Run to output a stream of build events in a machine-parseable form.
+use crate::build_loop::Event;
+use crate::ops::{
+    error::{ok, ExitError, OpResult},
+    get_paths,
+};
+use crate::rpc;
+use crossbeam_channel::{select, unbounded};
+use slog_scope::debug;
+use std::convert::TryInto;
+use std::str::FromStr;
+use std::thread;
+use std::time::Duration;
+
+/// Options for the kinds of events to report
+#[derive(Debug)]
+pub enum EventKind {
+    /// Report only live events - those that happen after invocation
+    Live,
+    /// Report events recorded for projects up until invocation
+    Snapshot,
+    /// Report all events
+    All,
+}
+
+impl FromStr for EventKind {
+    type Err = String;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        match s {
+            "all" => Ok(EventKind::All),
+            "live" => Ok(EventKind::Live),
+            "snapshot" => Ok(EventKind::Snapshot),
+            _ => Err(format!("{} not in all,live,snapshot", s)),
+        }
+    }
+}
+
+#[derive(Debug)]
+enum Error {
+    Varlink(rpc::Error),
+    Compat(String),
+}
+
+/// See the documentation for lorri::cli::Command::Shell for more
+/// details.
+pub fn main(kind: EventKind) -> OpResult {
+    // TODO: set up socket path, make it settable by the user
+    debug!("Starting stream_events");
+    let address = get_paths()?.daemon_socket_address();
+
+    use rpc::VarlinkClientInterface;
+    let mut client = rpc::VarlinkClient::new(
+        varlink::Connection::with_address(&address).expect("failed to connect to daemon server"),
+    );
+
+    let mut snapshot_done = false;
+    let (tx, rx) = unbounded();
+    let recycle = tx.clone();
```

Comment:
`tx_clone` perhaps? `recycle` as a name doesn't carry much intuition, at least for me.

Reply:
I was hoping to indicate the reason for this channel - to feed a first event back into our local channel that's tee'd from the daemon monitor.

Reply:
That's fine. I find naming channels particularly difficult - I find myself looking at their uses a lot to figure out what they're actually meant to do. You _could_ document this channel's purpose in a comment if you think that would be useful and possible to do succinctly. If not, don't - your call :)